ci/test/03_test_script.sh

```diff
@@ -33,7 +33,9 @@ echo "=== BEGIN env ==="
 env
 echo "=== END env ==="
 
-(
+if [ "$RUN_TIDY" != "true" ]; then
+  # The tidy task has no UB detector, so skip the patch because it is not
+  # needed and to avoid issues later on in the task.
```

Comment:
In commit "ci: Do not patch `leveldb` to workaround UB in "tidy" CI job" (b495e2dc95db42e3219720f5beeb6c136d067a4a)

Could the comment clarify what the issues are? Even just a hint or pointer to more information would be helpful. if they are IWYU issues, maybe just say "to avoid IWYU issues"